lib/resourceapply/interface.go

```diff
@@ -0,0 +1,16 @@
+package resourceapply
+
+import (
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CreateOnlyAnnotation means that this resource should be created if it does not exist, but should not be updated
+// if it already exists.  It is uniformly respected across all resources, but the first known use-cases are for
+// empty config.openshift.io and initial low-level operator resources.
+const CreateOnlyAnnotation = "release.openshift.io/create-only"
+
+func IsCreateOnly(metadata metav1.Object) bool {
+	return strings.EqualFold(metadata.GetAnnotations()[CreateOnlyAnnotation], "true")
```

Comment:
is there a guarantee that `GetAnnotations` returns non-nil map?

Reply:
Gets on nil maps are allowed

Reply:
> is there a guarantee that `GetAnnotations` returns non-nil map?

retrieval from a nil map is the zero value of the values in that map which works correctly here.